### PR TITLE
rgw: fix URI setting related issue

### PR DESCRIPTION
**Pull request by mikulely**
@rzarzynski @cbodley The origin meaning of the `FIXME` is to replace `uri` with `local_uri` ?

### CODE REVIEW

File: src/rgw/rgw_civetweb.cc [hunk 1]
```diff
@@ -112,7 +112,7 @@ void RGWCivetWeb::init_env(CephContext *cct)
 
   env.set("REQUEST_METHOD", info->request_method);
   env.set("REQUEST_URI", info->request_uri); // get the full uri, we anyway handle abs uris later
-  env.set("SCRIPT_URI", info->uri); /* FIXME */
+  env.set("SCRIPT_URI", info->local_uri);
```
Comment (theanalyst) [inline src/rgw/rgw_civetweb.cc]: uri is deprecated iirc, +1 for the change

File: src/rgw/rgw_asio_client.cc [hunk 2]
```diff
@@ -66,14 +66,14 @@ void ClientIO::init_env(CephContext *cct)
...
-  auto url = boost::string_ref{request.url};
-  auto pos = url.find('?');
-  auto query = url.substr(pos + 1);
-  url = url.substr(0, pos);
+  auto uri = boost::string_ref{request.url};
+  auto pos = uri.find('?');
+  auto query = uri.substr(pos + 1);
+  uri = uri.substr(0, pos);
 
-  env.set("REQUEST_URI", url);
```
Comment (mikulely) [inline src/rgw/rgw_asio_client.cc]: Or this's intended for forward request?

**Comment by mikulely** [comment]
Jenkins, retest this please.

**Comment by joscollin** [comment]
Jenkins, retest this please.

File: src/rgw/rgw_asio_client.cc (hunk 2 above)
Comment (mattbenjamin) [inline src/rgw/rgw_asio_client.cc]: pretty sure it's not intended to do that

Comment (mikulely) [inline src/rgw/rgw_asio_client.cc]: Thanks for confirmation. ;-)

**Comment by joscollin** [comment]
Test Result:

- http://pulpito.front.sepia.ceph.com/jcollin-2017-08-11_03:03:58-rgw-wip-jcollin-testing_08-10-2017-distro-basic-smithi/
- http://pulpito.front.sepia.ceph.com/compare/?branch=wip-jcollin-testing_08-10-2017&suite=rgw

**Comment by joscollin** [comment]
Jenkins retest this please

**Comment by mikulely** [comment]
ping

**Comment by yuriw** [comment]
@mikulely pls fix/rebase

--- pr 16935 --- pulling https://github.com/mikulely/ceph.git branch uri-fixme-cleanup
remote: Counting objects: 10, done.
remote: Total 10 (delta 8), reused 8 (delta 8), pack-reused 2
Unpacking objects: 100% (10/10), done.
From https://github.com/mikulely/ceph
 * branch            uri-fixme-cleanup -> FETCH_HEAD
Auto-merging src/rgw/rgw_asio_client.cc
CONFLICT (content): Merge conflict in src/rgw/rgw_asio_client.cc
CONFLICT (file/directory): There is a directory with name src/dmclock in f36cb5b7732a5439415a3ae3600a2fa51e21452e. Adding src/dmclock as src/dmclock~HEAD
Adding qa/suites/rgw/verify/frontend
Adding qa/suites/rgw/multisite/frontend
Adding qa/suites/rbd/mirror/cluster
Adding qa/suites/rbd/mirror/base
Adding qa/suites/ceph-deploy/basic/distros/ubuntu_latest.yaml
Adding qa/suites/ceph-deploy/basic/distros/centos_latest.yaml
Adding qa/suites/ceph-deploy/basic/distros/.qa
warning: inexact rename detection was skipped due to too many files.
warning: you may want to set your merge.renamelimit variable to at least 2076 and retry the command.
Automatic merge failed; fix conflicts and then commit the result.
Traceback (most recent call last):
  File "/home/yuriw/wip_master/src/script/build-integration-branch", line 62, in <module>
    assert not r
AssertionError

**Pull request closed without merging**